File: apps/mobile/lib/features/supplier/models/supplier_service_focus.dart
/// Catalog of supplier service-focus areas used in onboarding and settings.
class SupplierServiceFocus {
  const SupplierServiceFocus({
    required this.key,
    required this.label,
    required this.description,
  });

  final String key;
  final String label;
  final String description;
}

/// Fixed catalog — keep keys in sync with server SUPPLIER_SERVICE_FOCUS_KEYS.
abstract final class SupplierServiceFocusCatalog {
  static const List<SupplierServiceFocus> all = [
    SupplierServiceFocus(
      key: 'signages',
      label: 'Signages',
      description: 'Indoor/outdoor signs, storefronts, acrylic, metal',
    ),
    SupplierServiceFocus(
      key: 'tarpaulins',
      label: 'Tarpaulins',
      description: 'Events, banners, large-format outdoor prints',
    ),
    SupplierServiceFocus(
      key: 'document_printing',
      label: 'Document Printing',
      description: 'Reports, theses, booklets, office documents',
    ),
    SupplierServiceFocus(
      key: 'apparel',
      label: 'Apparel / Shirt Printing',
      description: 'DTG, sublimation, heat press, uniforms',
    ),
    SupplierServiceFocus(
      key: 'stickers_labels',
      label: 'Stickers & Labels',
      description: 'Die-cut, vinyl, product labels, packaging',
    ),
    SupplierServiceFocus(
      key: 'large_format',
      label: 'Large Format',
      description: 'Posters, wall murals, vehicle wrap panels',
    ),
    SupplierServiceFocus(
      key: '3d_printing',
      label: '3D Printing',
      description: 'Prototypes, models, custom parts',
    ),
    SupplierServiceFocus(
      key: 'invitations_cards',
      label: 'Invitations & Cards',
      description: 'Wedding, events, calling cards, specialty paper',
    ),
  ];

  static SupplierServiceFocus? byKey(String key) {
    for (final item in all) {
      if (item.key == key) return item;
    }
    return null;
  }

  static String labelFor(String key) => byKey(key)?.label ?? key;
}

File: admin/src/utils/supplier-service-focus.ts
/** Keep keys in sync with server SUPPLIER_SERVICE_FOCUS_KEYS. */
export const SUPPLIER_SERVICE_FOCUS_LABELS: Record<string, string> = {
  signages: "Signages",
  tarpaulins: "Tarpaulins",
  document_printing: "Document Printing",
  apparel: "Apparel / Shirt Printing",
  stickers_labels: "Stickers & Labels",
  large_format: "Large Format",
  "3d_printing": "3D Printing",
  invitations_cards: "Invitations & Cards",
};

export type RankedServiceFocus = {
  rank: number;
  key: string;
  label: string;
};

export function rankLabel(rank: number): string {
  if (rank === 1) return "Top 1";
  if (rank === 2) return "Top 2";
  if (rank === 3) return "Top 3";
  return `Top ${rank}`;
}

export function labelForServiceFocusKey(key: string): string {
  return SUPPLIER_SERVICE_FOCUS_LABELS[key] ?? key.replace(/_/g, " ");
}

export function toRankedServices(
  ranks: string[] | null | undefined,
): RankedServiceFocus[] {
  if (!Array.isArray(ranks)) return [];
  const seen = new Set<string>();
  const out: RankedServiceFocus[] = [];
  for (const raw of ranks) {
    const key = String(raw ?? "").trim();
    if (!key || seen.has(key)) continue;
    seen.add(key);
    out.push({
      rank: out.length + 1,
      key,
      label: labelForServiceFocusKey(key),
    });
  }
  return out;
}
